packages/core/src/util/resolvers.ts

```diff
@@ -59,6 +59,17 @@ export const resolveData = (instance: any, dataPath: string): any => {
     }, instance);
 };
 
+export const resolveArraySchema = (schema: JsonSchema,
```

Comment:
We have to be careful with this method. It not only resolves something, but it actually creates a new object instance. So when it's used in some `mapState...` or in a renderer it will break memoization and lead to a full rerender every cycle.

At the moment it's used in `resolveSubSchemas` which already returns a new instance and therefore has the same problem. I guess we missed that originally as the combinator renderers (where it is used) are not that common. However this `resolveArraySchema` is now also called in `mapStateToControlProps` and will therefore break memoization for all schemas which have arrays whose items need to be resolved.

It's such a special case of resolving that I would not offer a special method for it. So we should remove `resolveArraySchema` but just handle it where it's needed:

- The testers for array items need to resolve themselves (which they can do when we give them the rootSchema)
- The `resolveSubSchemas` is actually not needed at all and can be removed completely. Instead the resolving can be done in `createCombinatorRendererInfos` and in `mapStateToCombinatorRendererProps` at the point of time it's needed.
- The resolving in `mapStateToControlProps` needs to be removed again and placed where it's needed, e.g. in `mapStateToArrayControlProps` where we can perform the item resolving.